### PR TITLE
fix: Improve disconnected device experience

### PR DESCRIPTION
**Pull request by DaveTryon**
#### Details

Improve our disconnected device scenario. As called out in #3938, when the device connection is removed, ADB automatically removes port forwarding, which breaks assumptions in the reconnect code (which pre-dates the auto-configuration of port forwarding).

The PR basically consists of 3 pieces:
1. We already try to remove previously configured port forwarding, but we don't account for the error that gets thrown if we're removing a non-existent port. Fix this.
2. The notion of the "Start over" button is doomed to fail if we've really lost the connection, because we've lost our port forwarding and can only restore it by going through the device setup steps. I've updated the disconnected dialog to remove the "Start over" button.
3. When the user clicks the "Connect a new device" button, we return to the device setup view. Its state is still on the "start testing" page, which is doomed to fail because it's after the port forwarding step. I've changed the code to reset the state to the "connect-device" step when the users presses "Start testing", so there's always a consistent experience for reconnecting. To better reflect the workflow, I changed the "Connect a new device" button (in the disconnected dialog) to "Re-detect device", then we let the user connect an appropriate device before proceeding. @juchampa has reviewed and approved the user experience.

##### Motivation

Address #3938.

##### Screenshots

Connection dialog 
Before | After
--- | ---
image | image

Destination dialog after clicking through (these dialogs are unchanged, but shown purely to convey the user experience--note that the old experience landed in the "start testing" step with all previous information intact, while the new experience lands in the "connect your device" step):
Before | After
--- | ---
image | image

##### Context

This assumes that we're really disconnected when we display the device disconnected dialog. If the service were to throw an error (as opposed to being disconnected), then it's overkill to reconnect the device. I'm not aware of any user feedback to suggest that we're getting service errors, but I'd argue for a separate error handler dialog if/when we feel the need to make the distinction between the 2 error conditions.

I experimented with auto-detecting the newly connected device when the user clicked from the disconnected device dialog, but it added some unfortunate coupling between the results view (where the disconnected dialog is hosted) and the setup action creator. Always returning to the "Connect a device" step avoided this. Users may need to click 1 more button, but it lets us leverage an existing esperience.

#### Pull request checklist

- [x] Addresses an existing issue: #3938
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [x] (UI changes only) Added screenshots/GIFs to description above
- [x] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/electron/platform/android/setup/steps/configuring-port-forwarding.ts [hunk 1]
```diff
@@ -1,19 +1,36 @@
...
+const removeOldForwardedPort = async (
+    deps: AndroidSetupDeps,
+    store: AndroidSetupStoreCallbacks,
+) => {
+    const existingPort = store.getScanPort();
+    if (existingPort) {
+        try {
+            deps.logger.log(`removing old tcp:${existingPort} forwarding`);
+            await deps.removeTcpForwarding(existingPort);
+            store.setScanPort(null);
```
Comment (RobGallo) [inline src/electron/platform/android/setup/steps/configuring-port-forwarding.ts:17]: Don't we want to set the scan port to null regardless of whether or not an exception was thrown?

File: src/tests/unit/tests/electron/views/device-disconnected-popup/device-disconnected-popup.test.tsx [hunk 2]
```diff
@@ -18,32 +18,18 @@ describe('DeviceDisconnectedPopup', () => {
     });
 
     describe('user interaction', () => {
-        it('handles on connect a new device', () => {
-            const onConnectNewDeviceMock: () => void = jest.fn(() => {});
+        it('handles on redetect device', () => {
+            const onRedecteDeviceMock: () => void = jest.fn(() => {});
```
Comment (RobGallo) [inline src/tests/unit/tests/electron/views/device-disconnected-popup/device-disconnected-popup.test.tsx:22]: Spelling: "onRedect"

File: src/electron/platform/android/setup/steps/configuring-port-forwarding.ts (hunk 1 above)
Comment (DaveTryon) [reply src/electron/platform/android/setup/steps/configuring-port-forwarding.ts]: Yup. Moved after the try/catch block

File: src/tests/unit/tests/electron/views/device-disconnected-popup/device-disconnected-popup.test.tsx (hunk 2 above)
Comment (DaveTryon) [reply src/tests/unit/tests/electron/views/device-disconnected-popup/device-disconnected-popup.test.tsx]: Fixed

**Review by RobGallo — APPROVED** [review]

**Pull request merged**